File: features/lead/src/main/java/com/nivasafinance/features/lead/exception/LeadValidationException.java
package com.nivasafinance.features.lead.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import com.nivasafinance.common.exception.ValidationException;
import org.springframework.context.MessageSource;

import java.io.Serial;

public class LeadValidationException extends ValidationException {

    @Serial
    private static final long serialVersionUID = 2938475619283746502L;

    public LeadValidationException(String messageKey, Object[] args, MessageSource messageSource) {
        super(ExceptionUtils.INSTANCE.createLocalizedMessage(messageKey, args, messageSource));
    }
}

File: features/lead/src/main/java/com/nivasafinance/features/lead/exception/LeadValidationException.kt
package com.nivasafinance.features.lead.exception

import com.nivasafinance.common.exception.ExceptionUtils
import com.nivasafinance.common.exception.ValidationException
import org.springframework.context.MessageSource
import java.math.BigDecimal
import java.util.UUID

open class LeadValidationException(
    messageKey: String,
    args: Array<Any>? = null,
    messageSource: MessageSource
) : ValidationException(
    ExceptionUtils.createLocalizedMessage(messageKey, args, messageSource)
)

